packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -306,9 +316,12 @@ class AppDomain extends Domain {
     }
   }
 
-  Future<Device> _getDevice(String deviceId) async {
-    List<Device> devices = await daemon.deviceDomain.getDevices();
-    return devices.firstWhere((Device device) => device.id == deviceId, orElse: () => null);
+  Future<List<Map<String, dynamic>>> discover(Map<String, dynamic> args) async {
+    Device device = await _getDevice(daemon, args);
+    List<DiscoveredApp> apps = await device.discoverApps();
+    return apps.map((DiscoveredApp app) =>
+      <String, dynamic>{'id': app.id, 'observatoryPort': app.observatoryPort}
```

Comment:
perhaps also include information here about whether the port is a host port or a device port (about whether it will need forwarding or not)? the iOS simulator runs on the host and doesn't need port forwarding